pkg/localnet/localcluster.go

```diff
@@ -0,0 +1,601 @@
+}
+
+// Stops local cluster [clusterName]
+func LocalClusterStop(
+	app *application.Avalanche,
+	clusterName string,
+) error {
+	networkDir := GetLocalClusterDir(app, clusterName)
+	return TmpNetStop(networkDir)
+}
+
+// Removes local cluster [clusterName]
+// First stops it if needed
+func LocalClusterRemove(
+	app *application.Avalanche,
+	clusterName string,
+) error {
+	if clusterName == "" {
+		return fmt.Errorf("invalid cluster '%s'", clusterName)
+	}
+	networkDir := GetLocalClusterDir(app, clusterName)
+	if !sdkutils.DirExists(networkDir) {
+		return fmt.Errorf("cluster directory %s does not exist", networkDir)
+	}
+	_ = LocalClusterStop(app, clusterName)
+	return os.RemoveAll(networkDir)
+}
+
+// Indicates if local cluster is running
+func LocalClusterIsRunning(app *application.Avalanche, clusterName string) (bool, error) {
+	networkDir := GetLocalClusterDir(app, clusterName)
+	status, err := GetTmpNetRunningStatus(networkDir)
+	if err != nil {
+		return false, err
+	}
+	return status == Running, nil
+}
+
+// Indicates if local cluster is partially running (only some nodes are executing)
+// Useful for stop and destroy flows that need to accomplish the operation
+// regardless the cluster is operative
+func LocalClusterIsPartiallyRunning(app *application.Avalanche, clusterName string) (bool, error) {
+	networkDir := GetLocalClusterDir(app, clusterName)
+	status, err := GetTmpNetRunningStatus(networkDir)
+	if err != nil {
+		return false, err
+	}
+	return status != NotRunning, nil
+}
+
+// Indicates if the cluster [clusterName] is connected to a network of kind [networkModel]
+func LocalClusterIsConnectedToNetwork(
+	app *application.Avalanche,
+	clusterName string,
+	networkModel models.Network,
+) (bool, error) {
+	network, err := GetLocalCluster(app, clusterName)
+	if err != nil {
+		return false, err
+	}
+	networkID, err := GetTmpNetNetworkID(network)
+	if err != nil {
+		return false, err
+	}
+	return networkID == networkModel.ID, nil
+}
+
+// Returns the network model the local cluster given by [clusterName]
+func GetLocalClusterNetworkModel(
+	app *application.Avalanche,
+	clusterName string,
+) (models.Network, error) {
+	networkDir := GetLocalClusterDir(app, clusterName)
+	return GetNetworkModel(networkDir)
+}
+
+// Gets a list of clusters connected to local network that are also running
+func GetRunningLocalClustersConnectedToLocalNetwork(app *application.Avalanche) ([]string, error) {
+	return GetFilteredLocalClusters(app, true, models.NewLocalNetwork(), "")
+}
+
+// Gets a list of clusters that are running
+func GetRunningLocalClusters(app *application.Avalanche) ([]string, error) {
+	return GetFilteredLocalClusters(app, true, models.UndefinedNetwork, "")
+}
+
+// Gets a list of clusters filtered by running status, network model, and
+// validated blockchains
+func GetFilteredLocalClusters(
+	app *application.Avalanche,
+	running bool,
+	network models.Network,
+	blockchainName string,
+) ([]string, error) {
```

Comment:
Seems that this function is used whenever running = true, should we remove this from argument?

Reply:
I would rather prefer to keep the option available for the future. It is useful for debugging and dev tasks.